url270fc4f5ee_ykcilborw_Joust_tgz-pJ8-fri_patterns_interpreter_parsergenerator_examples_HowToOverrideLexerImplJ8/src/fri/patterns/interpreter/parsergenerator/util/SyntaxChecker.java

```diff
@@ -0,0 +1,139 @@
+package fri.patterns.interpreter.parsergenerator.util;
+
+import java.io.File;
+import java.util.*;
+import fri.patterns.interpreter.parsergenerator.Token;
+import fri.patterns.interpreter.parsergenerator.syntax.*;
+import fri.patterns.interpreter.parsergenerator.syntax.builder.*;
+
+/**
+	SyntaxChecker checks a EBNF syntax (file) for following things:
+	<ul>
+		<li>unresolved nonterminals (nonterminals without rule)</li>
+		<li>singular rules (nonterminal can be substituted by its singular right symbol)</li>
+		<li>isolated rules (redundant, can be removed)</li>
+		<li>None or more than one toplevel rule</li>
+	</ul>
+	<pre>
+	SYNTAX: java fri.patterns.interpreter.parsergenerator.util.SyntaxChecker file.syntax [file.syntax ...]
+	</pre>
+
+	@author (c) 2000, Fritz Ritzberger
+*/
+
+public class SyntaxChecker
+{
+	private boolean diagnosis = true;
+	
+	public SyntaxChecker(Object syntaxFile)
+		throws Exception
+	{
+		this(new SyntaxBuilder(syntaxFile).getSyntax());
+	}
+
+	public SyntaxChecker(Syntax syntax)	{
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 23. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 45-56, lines 59-68, lines 71-89, lines 92-107 into separate methods.